src/plugins/vis_augmenter/public/utils/utils.ts

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { get, isEmpty } from 'lodash';
+import { Vis } from '../../../../plugins/visualizations/public';
+import {
+  formatExpression,
+  buildExpressionFunction,
+  buildExpression,
+  ExpressionAstFunctionBuilder,
+} from '../../../../plugins/expressions/public';
+import {
+  ISavedAugmentVis,
+  SavedAugmentVisLoader,
+  VisLayerFunctionDefinition,
+  VisLayer,
+  isVisLayerWithError,
+  VisLayerErrorTypes,
+} from '../';
+import { PLUGIN_AUGMENTATION_ENABLE_SETTING } from '../../common/constants';
+import { getUISettings } from '../services';
+import { IUiSettingsClient } from '../../../../core/public';
+
+export const isEligibleForVisLayers = (vis: Vis, uiSettingsClient?: IUiSettingsClient): boolean => {
+  // Only support date histogram and ensure there is only 1 x-axis and it has to be on the bottom.
+  // Additionally to have a valid x-axis, there needs to be a segment aggregation
+  const hasValidXaxis =
+    vis.data?.aggs !== undefined &&
+    vis.data.aggs?.byTypeName('date_histogram').length === 1 &&
+    vis.params.categoryAxes.length === 1 &&
+    vis.params.categoryAxes[0].position === 'bottom' &&
+    vis.data.aggs?.bySchemaName('segment').length > 0;
+  // Support 1 segment for x axis bucket (that is date_histogram) and support metrics for
+  // multiple supported yaxis only. If there are other aggregation types, this is not
+  // valid for augmentation
+  const hasCorrectAggregationCount =
+    vis.data?.aggs !== undefined &&
+    vis.data.aggs?.bySchemaName('metric').length > 0 &&
+    vis.data.aggs?.bySchemaName('metric').length === vis.data.aggs?.aggs.length - 1;
+  const hasOnlyLineSeries =
+    vis.params?.seriesParams !== undefined &&
+    vis.params?.seriesParams?.every(
+      (seriesParam: { type: string }) => seriesParam.type === 'line'
+    ) &&
+    vis.params?.type === 'line';
+
+  // Checks if the augmentation setting is enabled
+  const config = uiSettingsClient ?? getUISettings();
+  const isAugmentationEnabled = config.get(PLUGIN_AUGMENTATION_ENABLE_SETTING);
+  return isAugmentationEnabled && hasValidXaxis && hasCorrectAggregationCount && hasOnlyLineSeries;
+};
+
+/**
+ * Using a SavedAugmentVisLoader, fetch all saved objects that are of 'augment-vis' type.
+ * Filter by vis ID.
+ */
+export const getAugmentVisSavedObjs = async (
+  visId: string | undefined,
+  loader: SavedAugmentVisLoader | undefined,
+  uiSettings?: IUiSettingsClient | undefined
+): Promise<ISavedAugmentVis[]> => {
+  // Using optional services provided, or the built-in services from this plugin
+  const config = uiSettings !== undefined ? uiSettings : getUISettings();
```

Comment:
I see 
```
const config = uiSettingsClient ?? getUISettings();
```
and
```
const config = uiSettings !== undefined ? uiSettings : getUISettings();
```
used in different places. While it would be a lot nicer to have just one of them, I would argue both are wrong and we should do
```
const config = uiSettings || getUISettings();
```